rapid_response_xblock/logger.py

```diff
@@ -2,18 +2,74 @@
 Capture events
 """
 from __future__ import absolute_import
+import logging
 
+from django.db import transaction
+from opaque_keys.edx.keys import UsageKey
+from opaque_keys.edx.locator import CourseLocator
 from track.backends import BaseBackend
 
+from rapid_response_xblock.models import ProblemCheckRapidResponse
+
+
+log = logging.getLogger(__name__)
+
 
 class LoggerBackend(BaseBackend):
     """
     Record events emitted by blocks.
     See TRACKING_BACKENDS for the configuration for this logger.
     For more information about events see:
 
-    http://edx.readthedocs.io/projects/devdata/en/stable/internal_data_formats/tracking_logs.html
+    http://edx.readthedocs.io/projects/devdata/en/stable/
+    internal_data_formats/tracking_logs.html
     """
 
     def send(self, event):
-        pass
+        # TODO: feature flag whitelisting valid problems
+        # TODO: only capture when problem is open
+        if event['event_type'] == 'problem_check':
+            try:
+                user_id = event['context']['user_id']
+                problem_id = UsageKey.from_string(
+                    event['event']['problem_id']
+                )
+                # usage_key.course_id may have a missing run
+                # so we need to grab the course key separately
+                course_key = CourseLocator.from_string(
+                    event['context']['course_id']
+                )
+
+                # This is assuming the event only includes information
+                # for one answer at a time
+                submission = event['event']['submission']
+                keys = list(submission.keys())
```

Comment:
Don't think the `list()` is necessary here

Reply:
It's needed for the `keys[0]` below

Reply:
If `submission` is a dict, `keys()` already returns a list. Is it not a dict?

```python
In [1]: type(dict(a=1, b=2).keys())
Out[1]: list
```

Reply:
Ah, I was thinking in Python 3 but you're right

    >>> dict(a=1, b=2).keys()
    dict_keys(['a', 'b'])
    >>> type(dict(a=1, b=2).keys())
    <class 'dict_keys'>

Reply:
Didn't know that about python 3. In light of that, I'd say leave this as-is